client/scripts/views/components/new_thread_form.ts

```diff
@@ -159,16 +160,37 @@ export const NewThreadForm: m.Component<{
     vnode.state.form = {};
     vnode.state.recentlyDeletedDrafts = [];
     vnode.state.uploadsInProgress = 0;
-    if (vnode.state.newType === undefined) {
-      vnode.state.newType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
+    vnode.state.overwriteConfirmationModal = false;
+    if (vnode.state.postType === undefined) {
+      vnode.state.postType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
     }
-    if (vnode.state.newType === PostType.Discussion) {
+    if (vnode.state.postType === PostType.Discussion) {
       vnode.state.form.threadTitle = localStorage.getItem(`${app.activeId()}-new-discussion-storedTitle`);
     } else {
       vnode.state.form.url = localStorage.getItem(`${app.activeId()}-new-link-storedLink`);
       vnode.state.form.linkTitle = localStorage.getItem(`${app.activeId()}-new-link-storedTitle`);
     }
   },
+  onremove: async (vnode) => {
```

Comment:
Would this be smoother if we used onbeforeremove?

Reply:
For whatever reason, onbeforeremove isn't hooking for me, have tried with several different configurations.